src/element/textElement.ts

```diff
@@ -261,32 +266,65 @@ const computeBoundTextPosition = (
 
 // https://github.com/grassator/canvas-text-editor/blob/master/lib/FontMetrics.js
 
-export const measureText = (text: string, font: FontString) => {
+export const measureText = (
+  text: string,
+  font: FontString,
+  lineHeight?: number,
+) => {
   text = text
     .split("\n")
     // replace empty lines with single space because leading/trailing empty
     // lines would be stripped from computation
     .map((x) => x || " ")
     .join("\n");
-
-  const height = getTextHeight(text, font);
+  const fontSize = parseInt(font);
+  const height = getTextHeight(text, fontSize, lineHeight);
   const width = getTextWidth(text, font);
 
   return { width, height };
 };
 
-const DUMMY_TEXT = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".toLocaleUpperCase();
-const cacheApproxLineHeight: { [key: FontString]: number } = {};
-
-export const getApproxLineHeight = (font: FontString) => {
-  if (cacheApproxLineHeight[font]) {
-    return cacheApproxLineHeight[font];
+export const computeNextLineHeightForText = (
+  originalElement: NonDeleted<ExcalidrawTextElement>,
+  updatedElement: NonDeleted<ExcalidrawTextElement>,
+) => {
+  const originalLineHeight = originalElement.lineHeight;
+  // Calculate line height relative to font size
+  if (originalLineHeight === originalElement.fontSize * 1.2) {
```

Comment:
```
if (originalLineHeight === originalElement.fontSize * 1.2) {
  return updatedElement.fontSize * 1.2;
}
```

I'm not comfortable with this.

1. does it always hold that a legacy lineHeight cannot be identical as `fontSize * 1.2` on some sizes? But, it may not matter as it likely wouldn't be noticed if we switch to a different algo at that point.
2. bigger worry is JS' floating points rounding errors. I'd do something like:
```
if (Math.abs(originalLineHeight - originalElement.fontSize * 1.2) < Number.EPSILON) {
```

All that said, if we know old `lineHeight` and current and old `height`, can't we just `(newHeight/oldHeight)oldFontSize`?

Reply:
I agree about floating-point rounding errors.

Reply:
Should we define `1.2` in `src/constants.ts`?  I see `1.2` in several places.

Reply:
yep, was actually discussed off-github. We'll be changing this to `1.25` to align with most browsers and our legacy algo, and moved to a constant, yes.